src/test/java/io/strimzi/kafka/bridge/http/ProducerIT.java

```diff
@@ -1117,4 +1122,98 @@ void sendSimpleMessageWithWrongContentType(VertxTestContext context) throws Inte
+                Stream.of(contentTypes)
+                    .filter(secondContentType -> !firstContentType.equals(secondContentType))
+                    .map(secondContentType -> Arguments.of(firstContentType, secondContentType)));
+
+    }
+
+    @ParameterizedTest
+    @MethodSource("contentTypeCombinations")
+    void dynamicContentTypeHandling(String firstContentType, String secondContentType, VertxTestContext context) throws Exception {
+        final String key = "exampleKey";
+        final String value = "Hello, world!";
+
+        JsonObject record = createDynamicRecord(key, value, firstContentType);
+        JsonArray records = new JsonArray().add(record);
+        JsonObject root = new JsonObject().put("records", records);
+
+        LOGGER.info("Sending:\n{}", root.toString());
+
+        // Send the first request
+        producerService()
+            .sendRecordsRequest(topic, root, firstContentType)
+            .sendJsonObject(root, response -> {
+                if (response.succeeded()) {
+                    HttpResponse<JsonObject> httpResponse = response.result();
+                    context.verify(() -> {
+                        assertThat(httpResponse.statusCode(), is(HttpResponseStatus.OK.code()));
+                        LOGGER.info("Successfully processed record with content type: {}", secondContentType);
+                    });
+                } else {
+                    context.failNow(response.cause());
+                }
+            });
+
+        record = createDynamicRecord(key, value, secondContentType);
+        records = new JsonArray().add(record);
+        root = new JsonObject().put("records", records);
+
+        // Send the second request
+        producerService()
+            .sendRecordsRequest(topic, root, secondContentType)
+            .sendJsonObject(root, response -> {
+                if (response.succeeded()) {
+                    HttpResponse<JsonObject> httpResponse = response.result();
+                    context.verify(() -> {
+                        assertThat(httpResponse.statusCode(), is(HttpResponseStatus.OK.code()));
+                        LOGGER.info("Successfully processed record with content type: {}", secondContentType);
+                        context.completeNow();
+                    });
+                } else {
+                    context.failNow(response.cause());
+                }
+            });
+
+        assertThat(context.awaitCompletion(TEST_TIMEOUT, TimeUnit.SECONDS), is(true));
+    }
+
+    private JsonObject createDynamicRecord(String key, String value, String contentType) {
+        JsonObject record = new JsonObject();
+
+        switch (contentType) {
+            case BridgeContentType.KAFKA_JSON_JSON:
+                // Both key and value are wrapped in a JsonObject
+                record.put("key", new JsonObject().put("actualKey", key));
+                record.put("value", new JsonObject().put("message", value));
```

Comment:
as I said in a previous comment you are passing JSON string not a full JSON object.
I mean, taking `value` as an example, you are passing something like `Hello, world!` which will be encoded as JSON string so including quotes like `"Hello, world!"`. But what if we want a test with `value` being something like this:

```json
{ "id": 10, "price": 150, "description": "Hello world" }
```

The above is a valid JSON value when embedded format `json` is used. The same is valid for key.

Reply:
Okay, I have updated to use `JSON` objects for all types and for `KAFKA_JSON_TEXT` just call `encode/decode` for string representation.